spmd/tensor/api.py

```diff
@@ -14,6 +15,79 @@
 """
 _DEBUG_STRICT = False
 
+# NOTE [Autograd interaction between torch.Tensor]
+#
+# The autograd functions defined below are being used by the public
+# facing APIs (i.e. from_local, local_tensor) to ensure our Tensor
+# works together with torch.Tensor within autograd engine. This
+# allows DistributedTensor to exist on part of the module hierarchy
+# and still able to calculate gradients across the torch.Tensor and
+# DistributedTensor boundary.
+# As an example, it enables backward computation of the following:
+#   Module A (with all torch.Tensor params)
```

Comment:
Would it be possible to use more concrete examples here? In the general case I feel like the conversion from local to distributed could lead to surprising behavior in the backward pass depending on the use case. 

For example --
1) if the user wraps a DDP _activation_ with DistributedTensor  , they should pass [Shard(0)] into the constructor. Then in the backward pass, the inverse operation would guarantee that we get a gradient that is also [Shard(0)]
2) however, if the user wraps a DDP _parameter_ with DistributedTensor, then they should pass [Replicate] since all ranks should have the same parameter. In the backward pass, we would potentially end up with a [Partial_] gradient, which we would then convert to Replicate -- but this is wrong, since DDP will later do an all-reduce again on an already replicated parameter.